v3/internal/utilization/aws.go

```diff
@@ -56,7 +77,18 @@ func getAWS(client *http.Client) (ret *aws, err error) {
 		}
 	}()
 
-	response, err := client.Get(awsEndpoint)
+	// AWS' IMDSv2 requires us to get a token before requesting metadata.
+	awsToken, err := getAWSToken(client)
+	if err != nil {
+		ret = nil
+		err = unexpectedAWSErr{e: fmt.Errorf("error contacting AWS IMDSv2 token endpoint, %v", err)}
```

Comment:
We should return here, if an error happened, right?
```suggestion
		return nil, unexpectedAWSErr{e: fmt.Errorf("error contacting AWS IMDSv2 token endpoint, %v", err)}
```

Reply:
Or, maybe you could create a request to `awsEndpoint` before calling `getAWSToken` and add the token to the request's headers only if `err != nil`. This would make IMDSv2 optional in case someone wants to stick with IMDSv1 (if that is even possible).

Reply:
Thanks for noticing this! After thinking harder about it, we shouldn't throw an unExpectedAWSErr at all, we're expecting a timeout unless they're in AWS. Fixing this and the test.

A more thorough implementation would raise errors on other HTTP errors, but I'm deferring that work in the interest of time.

Reply:
Regarding the extra request to awsEndpoint, I'd rather avoid making three calls to AWS and increasing the overhead by another 0.5 second on startup. As of now, IMDSv2 endpoints are always available -- the only control the customer has is to turn IMDSv1 off to help their security posture.